### ISSUE TITLE
Sync issue

### ISSUE THREAD

**Issue by BenoitCastang** [jpc-ae/Snippets#2]
Hi,
I'm using your extension on a day to day basis as a helpdesk technician and I love it !
I ve got a problem though is I cant manage to sync my snippets. Export works fine and the json looks good but the import doesnt give any output ?
Got any idea of a solution ?
Thanks in advance

**Comment by jpc-ae** [comment]
I wonder if a bug I discovered in my dev version snuck into the current
beta, where the snippets data ended up stringified on export. I'll
double-check and try to push a fix soon. In the meantime, if you're looking
at the json file and see a string next to space.data, try removing the
quotes and unescaping any contained quotes. I'm also happy to take a look
at it myself with an agreement that I'll delete the data after fixing and
sending it back to you.

Regards,
JPC

**Comment by BenoitCastang** [comment]
Thanks for your quick answer !
I tried with a simplest example and it still fails:
[snippets-backup-2023-11-29T20_22.json](https://github.com/jpc-ae/Snippets/files/13505185/snippets-backup-2023-11-29T20_22.json)

**Comment by jpc-ae** [comment]
Hmm.. that should be working. I see the backup is set to not sync. Can you
try setting your current data to not sync before importing and see if that
fixes it? It might be another bug I quashed in the current dev version.

**Comment by BenoitCastang** [comment]
What do you mean the backup is set to not sync ?
I tried setting synced to true, is that what you meant ? It did not work by the way.
[snippets-backup-2023-11-29T20_22.json](https://github.com/jpc-ae/Snippets/files/13505509/snippets-backup-2023-11-29T20_22.json)

**Comment by BenoitCastang** [comment]
I also mention Im using Edge, maybe it makes a difference.

**Comment by jpc-ae** [comment]
Okay, I've confirmed it does work on my latest dev version, but something on v0.9.3 isn't. I think it's a race condition where it's not waiting to parse the data before reloading. It looks like the clippings compatible export is broken too somehow.

I was hoping to fill out my current version a bit more to handle custom fields, but it looks like I'll need to clean it up and push it early for you. Once I have a stable version, I'll let you know and you can load it unpacked if you need it immediately. Thanks for bringing this to my attention.

**Comment by BenoitCastang** [comment]
Thanks to you, keep up the good work !

**Comment by jpc-ae** [comment]
Thanks for the vote of confidence!

In the meantime, if you change the file manually so createdBy says `"Clippings"` instead of `"Snippets"`, the first `"children"` is instead `"userClippingsRoot"`, and then delete `"space":{"synced":<bool>,"name":"Snippets","data":{"version":"0.9","timestamp":<num>,` along with the final 2 `}}`, it should bypass the part of the code causing the issue. I've attached an example:
[snippets-backup-2023-11-29T21_30.json](https://github.com/jpc-ae/Snippets/files/13505766/snippets-backup-2023-11-29T21_30.json)

**Comment by BenoitCastang** [comment]
Thank you it worked !
Im a tweak away from being completely able to export my collection is I cant go with folders. I tried this but it did not work:
[snippets-backup-2023-11-30T16_07.json](https://github.com/jpc-ae/Snippets/files/13515204/snippets-backup-2023-11-30T16_07.json)

**Comment by jpc-ae** [comment]
I'm almost there with my cleanup so I can push the fix for review. Just a tweak or two left. If you can wait a week for the review to go through, everything should work correctly then.

**Comment by BenoitCastang** [comment]
No problem I can wait a week :)

**Comment by jpc-ae** [comment]
Unfortunately it's still in review, though it's live on Chrome. More unfortunately, I just discovered a bug I didn't manage to quash in my race to push this fix out. When editing snippets, the context menus will be gone until you reload the browser. It's another race condition where reloading the snippets starts before the new space has loaded. I've fixed it now and will push it immeditely once the current update makes it to you.
